doc/hardware/peripherals/stepper.rst

```diff
@@ -1,24 +1,64 @@
 .. _stepper_api:
 
-Stepper API
-###########
+Steppers
+########
 
-Overview
-********
+The stepper driver API provides a set of functions for controlling and configuring stepper drivers.
 
-The Stepper API provides a set of functions for controlling and configuring stepper motors.
-It supports a variety of operations, including enabling/disabling the controller, setting the
-target position of the motor and thereby setting the motor in motion, setting/getting the actual
-position of the motor and so on.
+Configure Stepper Driver
+========================
 
-Configuration Options
-*********************
+- **Enable** the stepper driver using :c:func:`stepper_enable`.
+- Configure **micro-stepping resolution** using :c:func:`stepper_set_micro_step_res`
+  and :c:func:`stepper_get_micro_step_res`.
+- Configure **actual position a.k.a step count** in microsteps using :c:func:`stepper_set_actual_position`
+  and :c:func:`stepper_get_actual_position`.
+- Set **max velocity** in micro-steps per second using :c:func:`stepper_set_max_velocity`
 
-Related configuration options:
+Control Stepper
+===============
 
-* :kconfig:option:`CONFIG_STEPPER_INIT_PRIORITY`
+- **Move by** +/- micro-steps a.k.a. **relative movement** using :c:func:`stepper_move`.
+- **Move to** a specific position a.k.a. **absolute movement** using :c:func:`stepper_set_target_position`.
+- Run incessantly with a **constant velocity** in a certain direction until
+  a stop is detected using :c:func:`stepper_enable_constant_velocity_mode`.
+- Check if the stepper is **moving** using :c:func:`stepper_is_moving`.
+
+Device Tree
+===========
+
+In the context of stepper controllers  device tree provides the initial hardware
+configuration for stepper drivers on a per device level. Each device must specify
+a device tree binding in Zephyr, and ideally, a set of hardware configuration options
+for things such as current settings, ramp parameters and furthermore. These can then
+be used in a boards devicetree to configure a stepper driver to its initial state.
+
+See examples in:
+
+- :dtcompatible:`adi,tmc5041`
+
+Discord
+=======
+
+Zephyr has a `stepper discord`_ channel for stepper related discussions, which
+is open to all.
```

Comment:
not convinced about adding this section tbh. Not that it is necessarily a bad idea, just that I'm not sure it makes sense to potentially duplicate the same kind of info in each driver/subsystem doc page rather than just assume people will find out about Discord on other more generalistic pages

Reply:
There are some pages that already have the discord link, that's why I thought of putting it over here as well. I think it would be good to have a shortcut like this.